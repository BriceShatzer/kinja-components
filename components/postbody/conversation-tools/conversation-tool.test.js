import * as React from 'react';
import { mount, shallow } from 'enzyme';

import ConversationTool, { Alert } from './conversation-tool';


describe('<ConversationTool />', () => {
	const onChange = jest.fn();
	const onReloadClick = jest.fn();
	const statuses = ['kitty', 'piggy', 'doggo'];

	const stubElement = (hasStatusChanged = false) => (
		<ConversationTool
			activeStatus={statuses[0]}
			footerContent={<p>footer</p>}
			hasStatusChanged={hasStatusChanged}
			locale='en-US'
			onChange={onChange}
			onReloadClick={onReloadClick}
			statuses={statuses}
			subtitle="pets are friends"
			title="pets"
			translatedStatuses={statuses}
		/>
	);


	it('should render properly', () => {
		const conversationTool = shallow(stubElement());
		expect(conversationTool).toMatchSnapshot();
	});

	it('should collapse box', () => {
		const liveblogControls = mount(stubElement());
		const chevronDown = liveblogControls.find('ChevronDown');
		chevronDown.simulate('click');

		expect(liveblogControls.state().isOpen).toBe(false);
	});

	it('should call onChange', () => {
		const conversationTool = mount(stubElement());
		const radio = conversationTool.find('Radio').at(1);
		radio.simulate('click');

		expect(onChange).toHaveBeenCalledTimes(1);
		expect(onChange).toHaveBeenCalledWith('piggy');
		onChange.mockClear();
	});

	it('should render Alert / callOnReloadClick', () => {
		const conversationTool = mount(stubElement(true));
		const alert = conversationTool.find(Alert);
		expect(alert.length).toBe(1);

		const reload = alert.find('Reload');
		reload.simulate('click');

		expect(onReloadClick).toHaveBeenCalledTimes(1);
		onReloadClick.mockClear();
	});
});




// });
