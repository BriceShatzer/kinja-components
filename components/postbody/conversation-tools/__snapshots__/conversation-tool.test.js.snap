// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ConversationTool /> should render properly 1`] = `
<EnsureDefaultTheme>
  <conversation-tool__Container
    activeChild={1}
    isOpen={true}
  >
    <conversation-tool__Title>
      pets
      : 
      <strong>
        kitty
      </strong>
    </conversation-tool__Title>
    <conversation-tool__Subtitle>
      pets are friends
    </conversation-tool__Subtitle>
    <conversation-tool__ButtonWrapper>
      <RadioGroup
        name="conversation-tool"
        onChange={[MockFunction]}
      >
        <Radio
          checked={true}
          key="radio-kitty"
          label="kitty"
          value="kitty"
        />
        <conversation-tool__Line
          active={false}
          key="line-kitty"
        />
        <Radio
          checked={false}
          key="radio-piggy"
          label="piggy"
          value="piggy"
        />
        <conversation-tool__Line
          active={false}
          key="line-piggy"
        />
        <Radio
          checked={false}
          key="radio-doggo"
          label="doggo"
          value="doggo"
        />
      </RadioGroup>
    </conversation-tool__ButtonWrapper>
    <conversation-tool__FooterContainer>
      <p>
        footer
      </p>
    </conversation-tool__FooterContainer>
    <conversation-tool__ChevronWrapper
      onClick={[Function]}
    >
      <ChevronDownIcon />
    </conversation-tool__ChevronWrapper>
  </conversation-tool__Container>
</EnsureDefaultTheme>
`;
